File: ts-sdk/src/sdk/models/operations/resenduserinvitation.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";


export enum ResendUserInvitationRequestBodyLanguageEnum {
    En = "en",
    De = "de"
}


export class ResendUserInvitationRequestBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=email" })
  email?: string;

  @SpeakeasyMetadata({ data: "json, name=language" })
  language?: ResendUserInvitationRequestBodyLanguageEnum;
}


export class ResendUserInvitationRequest extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: ResendUserInvitationRequestBody;
}


export class ResendUserInvitationResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  userV2?: shared.UserV2;
}

File: user-api/python-sdk/src/epilot/models/operations/resenduserinvitation.py
import dataclasses
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from epilot import utils
from ..shared import userv2 as shared_userv2

class ResendUserInvitationRequestBodyLanguageEnum(str, Enum):
    EN = "en"
    DE = "de"


@dataclass_json
@dataclasses.dataclass
class ResendUserInvitationRequestBody:
    email: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('email') }})
    language: Optional[ResendUserInvitationRequestBodyLanguageEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('language') }})
    

@dataclasses.dataclass
class ResendUserInvitationRequest:
    request: Optional[ResendUserInvitationRequestBody] = dataclasses.field(default=None, metadata={'request': { 'media_type': 'application/json' }})
    

@dataclasses.dataclass
class ResendUserInvitationResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    user_v2: Optional[shared_userv2.UserV2] = dataclasses.field(default=None)
    
